### PR TITLE
fix(e2ei): force login to idp to update certificate (WPB-6877)

### PR DESCRIPTION
**Pull request by AndroidBob**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6877" title="WPB-6877" target="_blank">BugWPB-6877</a>  [Android] Fix Update E2EI Certificate
  </td></table>
  <br />
 

Cherry pick from the original PR: 
- #2742

---- 

 ⚠️ Conflicts during cherry-pick:
app/src/main/kotlin/com/wire/android/ui/e2eiEnrollment/GetE2EICertificateUI.kt
kalium

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like 
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Since the update mechanism and design required some extra work, and we already have couple of suggestion about other approaches, we decided to enforce the user to login everytime agains the idp and not using the refresh token.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. .

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3351](https://10.10.124.134/job/AR-build-pipeline/3351/) **failed**.

**Comment by AndroidBob** [comment]
Build [3392](https://10.10.124.134/job/AR-build-pipeline/3392/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/e2eiEnrollment/GetE2EICertificateUI.kt [hunk 1]
```diff
@@ -43,7 +43,7 @@ fun GetE2EICertificateUI(
     LaunchedEffect(Unit) {
         viewModel.requestOAuthFlow.onEach {
             OAuthUseCase(context, it.target, it.oAuthClaims, it.oAuthState).launch(
-                context.getActivity()!!.activityResultRegistry
+                context.getActivity()!!.activityResultRegistry, forceLoginFlow = true
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/e2eiEnrollment/GetE2EICertificateUI.kt:46]: this is the only place where `OAuthUseCase.launch` is called  so the parameter `forceLoginFlow` is always true. What is the point of it then?

**Comment by AndroidBob** [comment]
Build [3424](https://10.10.124.134/job/AR-build-pipeline/3424/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9031901)(9031901)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9031901)(9031901)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Review by borichellow — APPROVED** [review]

**Pull request merged**